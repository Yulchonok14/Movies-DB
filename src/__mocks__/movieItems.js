export const movieItems = [
    {
        id: 111,
        title: 'Bloodshot',
        release_date: '2020-05-01',
        vote_average: 3.6,
        genres: ['Action', 'Drama', 'Sci-Fi'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/9/9d/Bloodshot_poster.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 110
    },
    {
        id: 222,
        title: 'The invisible man',
        release_date: '2019-05-01',
        vote_average: 4.6,
        genres: ['Horror', 'Sci-Fi', 'Thriller'],
        poster_path: 'https://avatars.mds.yandex.net/get-kinopoisk-image/1900788/a90dd0dd-a8ae-4112-9865-23eb8f5879a8/360',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 185
    },
    {
        id: 333,
        title: 'Monster trucks',
        release_date: '2017-06-01',
        vote_average: 3.9,
        genres: ['Adventure', 'Action', 'Comedy'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/5/5e/Monster_Trucks_%28film%29.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 154
    },
    {
        id: 444,
        title: 'Cold comes the night',
        release_date: '2018-05-01',
        vote_average: 4.4,
        genres: ['Drama', 'Thriller', 'Crime'],
        poster_path: 'https://laughingsquid.com/wp-content/uploads/2014/01/CCTN_poster_Billy_HD-for-web.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 129
    },
    {
        id: 555,
        title: 'Snowden',
        release_date: '2020-05-21',
        vote_average: 4.8,
        genres: ['Drama', 'Thriller', 'Biography'],
        poster_path: 'https://upload.wikimedia.org/wikipedia/ru/1/1a/Snowden_%28film%29.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 113
    },
    {
        id: 666,
        title: 'Knock knock',
        release_date: '2011-05-01',
        vote_average: 2.6,
        genres: ['Horror', 'Thriller', 'Adventure'],
        poster_path: 'https://i1.wp.com/morbidlybeautiful.com/wp-content/uploads/2015/10/kk-2.jpg',
        overview: 'Vin Diesel fans who can’t wait for the next installment of the “Fast and Furious” macho soap opera series can get their fix at “Bloodshot,” a comic book adaptation that’s as big a stickler about “family” yet far less satisfying than even the worst films of the "Fast" franchise. The family in question here is the wife of Ray Garrison (Diesel), who is put in danger by her spouse’s mercenary soldiering. Now, if you want to walk into director Dave Wilson’s sci-fi actioner as blindly as I did, exit this review now. If you desire a hint of what you’re in for, let me leave you with a few phrases you would have encountered had you stuck around: “Universal Soldier,” “robotic cucarachas,” “needle drop abuse of the Talking Heads” and “blatant rip-off.”',
        runtime: 187
    }
];