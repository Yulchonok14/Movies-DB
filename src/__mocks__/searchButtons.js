export const searchByButtonsArr = [
    {id: 111, text: 'title', label: 'title', isActive: true},
    {id: 222, text: 'genres', label: 'genre', isActive: false}
];