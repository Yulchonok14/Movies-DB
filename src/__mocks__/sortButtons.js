export const sortByButtonsArr = [
    {id: 111, text: 'release_date', label: 'release date', isActive: true},
    {id: 222, text: 'vote_average', label: 'rating', isActive: false}
];