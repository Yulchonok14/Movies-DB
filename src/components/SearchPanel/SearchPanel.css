.SearchPanel {
    padding: 60px 20px;
}

.SearchPanel span{
    font-size: 30px;
    color: #FFFFFF;
    text-transform: uppercase;
}

@media (min-width: 600px) {
    .SearchPanel {
        padding: 0 100px 80px;
    }

    .SearchPanel span{
        font-size: 40px;
    }
}