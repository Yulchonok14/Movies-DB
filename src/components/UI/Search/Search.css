.Search {
    display: flex;
    justify-content: space-between;
    align-items: center;
    box-sizing: border-box;
    flex-direction: column;
}

.Search input {
    color: #FFFFFF;
    background-color: #232323;
    width: 100%;
    height: 60px;
    border: none;
    text-transform: capitalize;
    font-size: 20px;
    padding-left: 20px;
    margin: 10px 0;
    outline: none;
    opacity: 0.8;
}

.Search input::placeholder {
    color: #555555;
}

.Search button {
    width: 230px;
    color: #FFFFFF;
    background-color: #F65261;
    border-radius: 4px;
    text-transform: uppercase;
    height: 60px;
    font-size: 20px;
    border: none;
    outline: none;
}

.Search button:hover {
    box-shadow: 9px 1px 11px rgba(192,192,192,192.85), 8px 10px 10px rgba(192,192,192,192.82);
}

@media (min-width: 600px) {
    .Search {
        flex-direction: row;
    }

    .Search input {
        width: 80%;
    }
}