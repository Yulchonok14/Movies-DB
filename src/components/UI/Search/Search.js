import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import classes from './Search.css';

const Search = (props) => {
    const inputRef = useRef(null);
    return (
        <div className={classes.Search}>
            <input ref={inputRef} placeholder="search" defaultValue={props.searchText}/>
            <button onClick={() => props.clicked(inputRef.current.value.toLowerCase())}>search</button>
        </div>
    )
};

Search.propTypes = {
    clicked: PropTypes.func.isRequired
};

export default Search;