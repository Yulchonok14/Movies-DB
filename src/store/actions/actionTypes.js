export const GET_MOVIES_HUNTER = 'GET_MOVIES_HUNTER';
export const SEARCH_FILTER = 'SEARCH_FILTER';
export const SORT_FILTER = 'SORT_FILTER';
export const SORT_MOVIES = 'SORT_MOVIES';

export const GET_MOVIES_DETAILS = 'GET_MOVIES_DETAILS';
export const CHOOSE_MOVIE_DETAILS = 'CHOOSE_MOVIE_DETAILS';
export const PASS_CHOSEN_MOVIE = 'PASS_CHOSEN_MOVIE';
