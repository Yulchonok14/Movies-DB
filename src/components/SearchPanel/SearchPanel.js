import React from 'react';
import PropTypes from 'prop-types';

import Search from '../UI/Search/Search';
import Filter from '../UI/Filter/Filter';
import classes from './SearchPanel.css';

const SearchPanel = (props) => (
    <div className={classes.SearchPanel}>
        <span>find your movie</span>
        <Search
            clicked={props.searchBtn}
        />
        <Filter
            clicked={props.switchSearchFilter}
            show={props.showSearch}
            label={props.label}
            buttons={props.searchByFilter}
        />
    </div>
);

SearchPanel.propTypes = {
    searchBtn: PropTypes.func.isRequired,
    switchSearchFilter: PropTypes.func.isRequired,
    showSearch: PropTypes.oneOfType([
        PropTypes.string.isRequired,
        PropTypes.bool.isRequired
        ]),
    label: PropTypes.string.isRequired,
    searchByFilter: PropTypes.arrayOf(PropTypes.shape({
        id:PropTypes.number.isRequired,
        text:PropTypes.string.isRequired,
        isActive: PropTypes.bool.isRequired
    }))
};

export default SearchPanel;