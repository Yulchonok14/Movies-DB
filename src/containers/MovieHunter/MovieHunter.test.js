import React from 'react';

import { configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk';


import ConnectedMovieHunter, { MovieHunter, mapDispatchToProps } from './MovieHunter';
import * as actionCreators from '../../store/actions/index';
import * as actionTypes from '../../store/actions/actionTypes';
import { movieItems } from '../../__mocks__/movieItems';
import { sortByButtonsArr } from '../../__mocks__/sortButtons';
import { searchByButtonsArr } from '../../__mocks__/searchButtons';
import SearchPanel from '../../components/SearchPanel/SearchPanel';
import Search from '../../components/UI/Search/Search';

configure({
    adapter: new Adapter()
});

describe('<MovieHunter/>', () => {
    const initialState = {
        hunter: {
            movieHunterItems: [],
            searchFilter: 'title',
            sortFilter: 'release_date'
        },
        searchFilterArr: searchByButtonsArr,
        sortFilterArr: sortByButtonsArr
    };
    const mockStore = configureStore([thunk]);
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<ConnectedMovieHunter store={store}/>);
    });

    it('render the connected component', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('check Prop matches with initialState', () => {
        expect(wrapper.find(MovieHunter).prop('movieItems')).toEqual(initialState.hunter.movieHunterItems);
        expect(wrapper.find(MovieHunter).prop('sortFilter')).toEqual(initialState.hunter.sortFilter);
        expect(wrapper.find(MovieHunter).prop('searchFilter')).toEqual(initialState.hunter.searchFilter);
    });

    it('test mapDispatchToProps', () => {
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).onSortFilterSwitched('vote_average');
        expect(dispatch.mock.calls[0][0]).toEqual({
            type: actionTypes.SORT_FILTER,
            payload: {sortFilterText: 'vote_average'}
        });

        mapDispatchToProps(dispatch).onMovieSearched('test');
        expect(dispatch.mock.calls[1][0]).toBeDefined();

        mapDispatchToProps(dispatch).onSortMovies();
        expect(dispatch.mock.calls[2][0]).toEqual({
            type: actionTypes.SORT_MOVIES
        });

        mapDispatchToProps(dispatch).onSearchFilterSwitched('title');
        expect(dispatch.mock.calls[3][0]).toEqual({
            type: actionTypes.SEARCH_FILTER,
            payload: {searchFilterText: 'title'}
        });

        mapDispatchToProps(dispatch).onMovieChosen(movieItems[0]);
        expect(dispatch.mock.calls[4][0]).toEqual({
            type: actionTypes.PASS_CHOSEN_MOVIE,
            payload: {chosenMovie: movieItems[0]}
        });
    });

    it('capturing Snapshot of MovieHunter', () => {
        const testFunction = () => true;
        const renderedValue = renderer.create(<MovieHunter
            movieItems={initialState.hunter.movieHunterItems}
            sortFilter={initialState.hunter.sortFilter}
            searchFilter={initialState.hunter.searchFilter}
            searchFilterArr={searchByButtonsArr}
            onMovieChosen={testFunction}
            sortFilterArr={sortByButtonsArr}/>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it('switchSearchFilterHandler dispatch SEARCH_FILTER action', () => {
        wrapper.find(MovieHunter).instance().switchSearchFilterHandler(0);
        const action = store.getActions();
        expect(action[0]).toEqual({
            type: actionTypes.SEARCH_FILTER,
            payload: {searchFilterText: 'title'}
        });
    });

    it('switchSortFilterHandler dispatch SORT_FILTER action', () => {
        let movieHunter = wrapper.find(MovieHunter).instance();
        movieHunter.switchSortFilterHandler(1);
        const action = store.getActions();
        expect(action[0]).toEqual({
                type: actionTypes.SORT_FILTER,
                payload: {sortFilterText: 'vote_average'}
            });
    });

    it('componentDidUpdate dispatch SORT_MOVIES action', () => {
        const prevProp = {sortFilter: 'vote_average'};
        wrapper.find(MovieHunter).instance().componentDidUpdate(prevProp);
        const action = store.getActions();
        expect(action[0]).toEqual({
            type: actionTypes.SORT_MOVIES
        });
    });

    it('check search button click', () => {
        let mockCallBack = jest.fn();
        wrapper = mount(<MovieHunter movieItems={movieItems} onMovieSearched={mockCallBack}/>);
        wrapper.find(SearchPanel).find(Search).find('button').simulate('click');
        expect(mockCallBack).toBeCalled();
    })
});