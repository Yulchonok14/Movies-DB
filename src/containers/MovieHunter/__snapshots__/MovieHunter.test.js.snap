// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieHunter/> capturing Snapshot of MovieHunter 1`] = `
Array [
  <div>
    <span>
      find your movie
    </span>
    <div>
      <input
        placeholder="search"
      />
      <button
        onClick={[Function]}
      >
        search
      </button>
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <label>
        search by
      </label>
      <div>
        <button
          onClick={[Function]}
        >
          title
        </button>
        <button
          className={null}
          onClick={[Function]}
        >
          genre
        </button>
      </div>
    </div>
  </div>,
  <div>
    <span>
      0 movie found
    </span>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <label>
        sort by
      </label>
      <div>
        <button
          onClick={[Function]}
        >
          release date
        </button>
        <button
          className={null}
          onClick={[Function]}
        >
          rating
        </button>
      </div>
    </div>
  </div>,
  <div>
    <div>
      No films found
    </div>
  </div>,
]
`;
